File: data/code_pairs/fortran/712.f90
module rrlw_kg03
    implicit none
    save

    ! Assuming kind parameter definitions for demonstration
    integer, parameter :: im = selected_int_kind(8)
    integer, parameter :: rb = selected_real_kind(p=6, r=37)

    ! Constants
    integer(kind=im), parameter :: no3 = 16
    real(kind=rb) :: fracrefao(no3,9), fracrefbo(no3,5)
    real(kind=rb) :: kao(9,5,13,no3)
    real(kind=rb) :: kbo(5,5,47,no3) ! Adjusted for 0-based indexing in tests
    real(kind=rb) :: kao_mn2o(9,19,no3), kbo_mn2o(5,19,no3)
    real(kind=rb) :: selfrefo(10,no3)
    real(kind=rb) :: forrefo(4,no3)

    integer(kind=im), parameter :: ng3 = 16
    real(kind=rb) :: fracrefa(ng3,9), fracrefb(ng3,5)
    real(kind=rb) :: ka(9,5,13,ng3)
    real(kind=rb) :: kb(5,5,47,ng3) ! Adjusted for 0-based indexing in tests
    real(kind=rb) :: ka_mn2o(9,19,ng3), kb_mn2o(5,19,ng3)
    real(kind=rb) :: selfref(10,ng3)
    real(kind=rb) :: forref(4,ng3)

end module rrlw_kg03

File: data/code_pairs/cpp/712.cpp
#include <array>

using rb = float; // Real(kind=rb) equivalent
using im = int;   // Integer(kind=im) equivalent

class rrlw_kg03 {
public:
    static constexpr im no3 = 16;
    static constexpr im ng3 = 16;

    std::array<std::array<rb, 9>, no3> fracrefao;
    std::array<std::array<rb, 5>, no3> fracrefbo;
    std::array<std::array<std::array<std::array<rb, no3>, 13>, 5>, 9> kao;
    std::array<std::array<std::array<std::array<rb, no3>, 47>, 5>, 5> kbo;
    std::array<std::array<std::array<rb, no3>, 19>, 9> kao_mn2o;
    std::array<std::array<std::array<rb, no3>, 19>, 5> kbo_mn2o;
    std::array<std::array<rb, no3>, 10> selfrefo;
    std::array<std::array<rb, no3>, 4> forrefo;

    std::array<std::array<rb, 9>, ng3> fracrefa;
    std::array<std::array<rb, 5>, ng3> fracrefb;
    std::array<std::array<std::array<std::array<rb, ng3>, 13>, 5>, 9> ka;
    std::array<std::array<rb, ng3>, 585> absa; // Flattened representation if needed
    std::array<std::array<std::array<std::array<rb, ng3>, 47>, 5>, 5> kb;
    std::array<std::array<rb, ng3>, 1175> absb; // Flattened representation if needed
    std::array<std::array<std::array<rb, ng3>, 19>, 9> ka_mn2o;
    std::array<std::array<std::array<rb, ng3>, 19>, 5> kb_mn2o;
    std::array<std::array<rb, ng3>, 10> selfref;
    std::array<std::array<rb, ng3>, 4> forref;
};